app/src/androidTest/java/com/example/concordia_campus_guide/InstrumentalTests/ViewModelFactoryInstrumentalTest.java

```diff
@@ -0,0 +1,43 @@
+package com.example.concordia_campus_guide.InstrumentalTests;
+
+import com.example.concordia_campus_guide.Activities.MainActivity;
+import com.example.concordia_campus_guide.Activities.RoutesActivityViewModel;
+import com.example.concordia_campus_guide.Activities.SearchActivityViewModel;
+import com.example.concordia_campus_guide.Fragments.InfoCardFragment.InfoCardFragment;
+import com.example.concordia_campus_guide.Fragments.InfoCardFragment.InfoCardFragmentViewModel;
+import com.example.concordia_campus_guide.Fragments.LocationFragment.LocationFragmentViewModel;
+import com.example.concordia_campus_guide.Helper.ViewModelFactory;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(AndroidJUnit4.class)
+public class ViewModelFactoryInstrumentalTest {
+    private ViewModelFactory viewModelFactory;
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Before
+    public void init() {
+        viewModelFactory = new ViewModelFactory(mActivityRule.getActivity().getApplication());
+    }
+
+    @Test
+    public void createTest(){
+        assertNotNull(viewModelFactory.create(SearchActivityViewModel.class));
```

Comment:
could you check assertTrue(viewModelFactory.create(SearchActivityViewModel.class) instanceof SearchActivityViewModel) instead?

also, could you pass a class that we would expect to return null, such as assertNull(viewModelFactory.create(MainActivity.class))?

Reply:
Yes i will adjust it that accordingly.

Reply:
It doesn't work with MainActivity because the class that we are passing needs to be extending ViewModel